src/document_basic.c

```diff
@@ -214,28 +262,59 @@ void Document_Free(Document *doc) {
   }
 }
 
-int Redis_SaveDocument(RedisSearchCtx *ctx, Document *doc, int options, QueryError *status) {
-  RedisModuleKey *k =
-      RedisModule_OpenKey(ctx->redisCtx, doc->docKey, REDISMODULE_WRITE | REDISMODULE_READ);
-  if (k == NULL || (RedisModule_KeyType(k) != REDISMODULE_KEYTYPE_EMPTY &&
-                    RedisModule_KeyType(k) != REDISMODULE_KEYTYPE_HASH)) {
-    QueryError_SetError(status, QUERY_EREDISKEYTYPE, NULL);
-    if (k) {
-      RedisModule_CloseKey(k);
-    }
-    return REDISMODULE_ERR;
+static RedisModuleString **globalAddRSstrings = NULL;
+
+static void initGlobalAddStrings() {
+  globalAddRSstrings = rm_malloc(3 * sizeof(*globalAddRSstrings));
```

Comment:
Why not put it on the static area?
#define globalAddRSstringsSize 3
static RedisModuleString *globalAddRSstrings[globalAddRSstringsSize]